src/java/org/apache/cassandra/service/reads/ReplicaFilteringProtection.java

```diff
@@ -270,6 +277,11 @@ public void onMergedRangeTombstoneMarkers(RangeTombstoneMarker merged, RangeTomb
 
                     @Override
                     public void close()
+                    {
+                    }
```

Comment:
should close call checkpoint also?

Reply:
I think, as it stands, there's no situation where calling `close()` on the merge listener would do anything useful, given we're manually checkpointing in `queryProtectedPartitions()`, which used to close the iterator before this patch (hence the checkpoint logic being in `close()`). Looking at the [other comment](https://github.com/apache/cassandra/pull/4150/files#r2085863141) below about whether we even want to modify the iterator contracts, let's come back to this...